src/main/java/it/gov/pagopa/rtd/ms/rtdmsfileregister/service/FileMetadataServiceImpl.java

```diff
@@ -115,4 +117,24 @@ public SenderAdeAckListDTO getSenderAdeAckList(List<String> senders) {
 
     return new SenderAdeAckListDTO(fileNameList);
   }
+
+  @Override
+  public FileMetadataDTO updateStatus(String filename, int status) {
+    FileMetadataEntity toBeUpdated = repository.findFirstByName(filename);
+
+    if (toBeUpdated == null) {
+      throw new FilenameNotPresent();
+    }
+
+    if (toBeUpdated.getStatus() == status) {
+      throw new StatusAlreadySet();
+    }
+
+    toBeUpdated.setStatus(status);
+
+    repository.removeByName(filename);
+
+    return modelMapper.map(repository.save(modelMapper.map(toBeUpdated, FileMetadataEntity.class)),
```

Comment:
is the inner mapping needed? `toBeUpdated` should be a `FileMetadataEntity` already

Reply:
Fixed, thank you!